### PR TITLE
[spec] Fix various typos in rules

### LINKED ISSUE #1316: Typo in runtime semantics of table.init
In the spec of [table.init](https://webassembly.github.io/spec/core/exec/instructions.html#xref-syntax-instructions-syntax-instr-table-mathsf-table-init-x-y) it is defined that `val=𝑆.𝖾𝗅𝖾𝗆𝗌[𝐹.𝗆𝗈𝖽𝗎𝗅𝖾.𝖾𝗅𝖾𝗆𝖺𝖽𝖽𝗋𝗌[𝑥]].𝖾𝗅𝖾𝗆[𝑠])`. It should be `elemaddrs[y]` instead.

### LINKED ISSUE #1319: [WAT Spec] Elem and Data single instructions abbreviations missing parens
The single-instruction abbreviations for both [elem](https://webassembly.github.io/spec/core/text/modules.html#id7) and [data](https://webassembly.github.io/spec/core/text/modules.html#id8) segments according to the spec should not be parenthesized, but in practice parsers like `wat2wasm` and `wasmtime` require the instructions be in parentheses.

I also noticed the implementations restrict the `offset` abbreviations to be either `i32.const` or `global.get`, and `item` abbreviations to be either `ref.null` or `ref.func`.

### PR DESCRIPTION
**Pull request by rossberg**
Fixes #1316, fixes #138, fixes #1319. Also fixes a number of other minor bugs.

### CODE REVIEW

File: test/core/data.wast [hunk 1]
```diff
@@ -382,6 +382,14 @@
   "type mismatch"
 )
 
+(assert_invalid
```
Comment (conrad-watt) [inline test/core/data.wast:385]: Is this the same test as in `elem.wast`? Should this instead be testing a data segment?

**Review by conrad-watt — COMMENTED** [review]
mostly LGTM aside from comment, although I'm not so familiar with the text format and would welcome a second opinion on those changes

File: test/core/data.wast (hunk 1 above)
Comment (rossberg) [reply test/core/data.wast:385]: Ah yes, indeed. Fixed.

**Comment by rossberg** [comment]
@laelath, could you take a look at the text format fixes and confirm that they fix the bugs you reported?

**Comment by laelath** [comment]
They look good to me!

**Review by conrad-watt — APPROVED** [review]

**Pull request merged**